go-controller/pkg/ovn/egressip.go

```diff
@@ -376,14 +376,25 @@ func (oc *Controller) reconcileEgressIPPod(old, new *v1.Pod) (err error) {
 			// If the podSelector is not empty then check if the pod stopped
 			// matching. If the pod was deleted, "new" will be nil and
 			// newPodLabels will not match, so this is should cover that case.
-			if !podSelector.Empty() && !podSelector.Matches(newPodLabels) && podSelector.Matches(oldPodLabels) {
-				return oc.deletePodEgressIPAssignments(egressIP.Name, egressIP.Status.Items, oldPod)
+			if !podSelector.Empty() && !podSelector.Matches(newPodLabels) {
+				if podSelector.Matches(oldPodLabels) {
+					err = oc.deletePodEgressIPAssignments(egressIP.Name, egressIP.Status.Items, oldPod)
+					if err != nil {
+						return err
+					}
+				}
+				// egressIP has a podSelector that pod does not match: skip it
```

Comment:
This comment is not aligned with the code. We will end up here even if the egress IP has a podSelector that matches the pod.